src/shared/sam/localLambdaRunner.ts

```diff
@@ -366,15 +366,16 @@ export async function runLambdaFunction(
         ...(config.aws?.region ? { AWS_DEFAULT_REGION: config.aws.region } : {}),
     }
 
-    const timer = createInvokeTimer(ctx.settings)
+    const buildTimer = createInvokeTimer(ctx.settings)
 
-    if (!(await buildLambdaHandler(timer, envVars, config))) {
+    if (!(await buildLambdaHandler(buildTimer, envVars, config))) {
         return config
     }
 
     await onAfterBuild()
 
-    if (!(await invokeLambdaHandler(timer, envVars, config))) {
+    const invokeTimer = createInvokeTimer(ctx.settings)
```

Comment:
Does it work instead to call `timer.refresh()` here?

Reply:
Telemetry data uses the timer's `elapsedTime` so better to refresh then to create a new one.